lib/index.js

```diff
@@ -29,8 +29,13 @@ function defaultFilename() {
 }
 
 function attachMethods(policy) {
-  policy.filter = function (vulns, root) {
-    return filter(vulns, policy, root || path.dirname(policy.__filename));
+  policy.filter = function (vulns, root, simple = false) {
```

Comment:
I'm not necessarily convinced that this interface is a good idea. A flag named after a behaviour that triggers a different code path is usually a bad smell. I'm not previously familiar with this part of our codebases, and as such I can't think of many other ideas at the moment.

One possible alternative would be to allow a function to be injected instead of a flag. Callers could then customise the actual filtering/matching behaviour using this function. One downside to that alternative is that in practice, that function would likely have to be duplicated in the CLI and registry code bases to support ignores in both of those contexts.

Reply:
if we can't think of anything "better" than behaviour-influencing parameters, and we don't want to inject the whole match function for the reason in my comment above, maybe we should change this to an enum-string, e.g. `strategy = 'simple' | 'package-manager'`, with `package-manager` being the default. That way, future use-cases can be added without breaking this interface.

Reply:
i'll b hone i love the strategy idea more since while reading this code... simple has no meaning... it's just a word right? but if we have it as a value for something that's called strategy and we pass matchStrategy around or whatever u call it that would make more sense right? 

what do u mean duplicate the function in CLI and registry? why would we want to do it? are we duplicating the simple logic u introduced here?

Reply:
@odinn1984 I totally agree, that variable name is not great. The concept I'm trying to express with "simple" is "exact matching of the path arrays, but also supporting *s". For example:

`foo > bar` matches ['foo', 'bar'].
' foo > *' matches ['foo', 'bar', 'baz']

This "relatively simple" behaviour is contrasted with the current default, which has a couple of package-manager-specific attributes:

1. Slicing the first element off the path before comparison
1. "fuzzy matching": If the path (minus the first element) "contains" the rule as a
substring, it will match. For example: the rule file.json will match
the path ["dir/file.json"].
1. parsing path elements as URLs or git objects and matching based on "normalised" forms of these

Perhaps the new, opt-in "simple" path can be named "exact", i.e. `matchStrategy = 'exact'`, vs the default of `matchStrategy = 'packageManager'`. "exact" isn't strictly correct because of the `*`s, but perhaps it's still better than "simple". suggestions welcome!

---

re: duplicate the function, that refers to my other possible idea, which is to optionally inject a match function instead of a strategy enum. The upside to that is it solves our naming problem somewhat, by inverting the behaviour into the calling code instead of here, but the downside is that if we want to call this policy library both in the CLI (for local-exec scans, as IaC supports) and also in registry to support server-side ignores, we'd need to duplicate that matching function.

Does this make sense?

Reply:
I've just pushed a commit to change the flag to a string, but I think the overall discussion about what interface to present is still open.